### PR TITLE
add __attribute__((unused)) for silent compiler warnings

### PR DESCRIPTION
**Pull request by jupe**
## Status
**READY**

## Migrations
NO

## Description
When compiling with reduced configuration compiler causes warnings by default because of unused function. This PR silents those warnings

## Related Issue
Fixes: #92 

## Todos
- [x] Tests

### CODE REVIEW

File: source/ns_cmdline.c [hunk 1]
```diff
@@ -126,6 +126,12 @@
 // end of default configurations
 #endif
 
+#if defined(__GNUC__) || defined(__clang__) || defined(__CC_ARM)
+#define CMDLINE_UNUSED __attribute__((unused))
```
Comment (jupe) [inline source/ns_cmdline.c]: https://github.com/ARMmbed/mbed-os/blob/a23a850a422e23531a260dd829bce9fd58aa9adf/platform/mbed_toolchain.h#L95 introduce similar attribute I wanted to keep cli as independent as possible and not rely on mbed-os macros. Not sure about what is differences between `__attribute__((unused))` and `__attribute__((__unused__))`. First one is documented [here](https://gcc.gnu.org/onlinedocs/gcc-3.2/gcc/Variable-Attributes.html).

Comment (jupe) [reply source/ns_cmdline.c]: @kjbracey-arm do you know difference? Also looks that this did not work for CI runs even it worked in my machine.🤔 needs more investigation..

Comment (jupe) [reply source/ns_cmdline.c]: I think I found the reason..

**Comment by jupe** [comment]
`build mbed-os3-frdm-k64f-armcc` fails because of missing ARMCC compiler in CI ([ISGTESTINF-1041](https://jira.arm.com/browse/ISGTESTINF-1041))

File: source/ns_cmdline.c (hunk 1 above)
Comment (kjbracey-arm) [reply source/ns_cmdline.c]: Fo

**Pull request merged**